### PR TITLE
Add retry if data not ready for SCD30

### PR DESCRIPTION
**Pull request by tyeth**
This updates the SCD30 driver to pull the data for all metrics in one go, rather than each one in turn which caused many data_ready wait issues - appearing as no data for most metrics.

### CODE REVIEW

**Review by tyeth — COMMENTED** [review]
@brentru ready for review.
This fixes #627

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_SCD30.h [hunk 1]
```diff
@@ -53,6 +53,36 @@ class WipperSnapper_I2C_Driver_SCD30 : public WipperSnapper_I2C_Driver {
...
+      @returns  True if the sensor was read successfully, False otherwise.
+  */
+  /*******************************************************************************/
+  bool readSensor() {
+    // dont read sensor more than once per second
+    if (_lastRead != 0 && millis() - _lastRead < 1000) {
+      return true;
+    }
+
+    if (!_scd->dataReady()) {
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_SCD30.h:68]: This segment is difficult to read. Could you refactor the logic out into to a new function?

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_SCD30.h [hunk 2]
```diff
@@ -64,14 +94,11 @@ class WipperSnapper_I2C_Driver_SCD30 : public WipperSnapper_I2C_Driver {
...
-      return false;
-
-    // attempt to get temperature data
-    sensors_event_t humidEvent;
-    if (!_scd->getEvent(&humidEvent, tempEvent))
+    if (!readSensor()) {
       return false;
+    }
 
+    tempEvent->temperature = _temperature;
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_SCD30.h:101]: I think it's confusing to mix types like this (`_temperature` is a `float`, `tempEvent` is a `sensors_event_t`). 

Instead, could we have:
*  `_temperature`, `_humidity` and `_co2` declared as protected `sensors_event_t` types?
* You could also remove the need for the (`sensors_event_t tempEvent;` and `sensors_event_t humidEvent;`) decls. on L77:L78. 
* You could then remove the (now redundant) reassignments in L79:L81, too.

**Review by brentru — CHANGES_REQUESTED** [review]
@tyeth  I agree with the overall structure, I do have some comments and questions about the types your using and how they're being packed.

**Review by brentru — APPROVED** [review]
This reads much better now, thanks for making adjustments.

**Pull request merged**